components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIMWSDLReader.java

```diff
@@ -152,6 +152,35 @@ public OMElement readAndCleanWsdl2(API api) throws APIManagementException {
 
     }
 
+    public byte[] getWSDL() throws APIManagementException {
+        try {
+            Definition wsdlDefinition = readWSDLFile();
+            WSDLWriter writer = getWsdlFactoryInstance().newWSDLWriter();
+            ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
+            writer.writeWSDL(wsdlDefinition, byteArrayOutputStream);
+            return byteArrayOutputStream.toByteArray();
+        } catch (Exception e) {
+            String msg = " Error occurs when change the addres URL of the WSDL";
+            log.error(msg);
+            throw new APIManagementException(msg, e);
+        }
+    }
+
+    public Definition getWSDLDefinitionFromByteContent(byte[] wsdl) throws APIManagementException {
+        try {
+            WSDLReader wsdlReader = getWsdlFactoryInstance().newWSDLReader();
+            // switch off the verbose mode
+            wsdlReader.setFeature(JAVAX_WSDL_VERBOSE_MODE, false);
+            wsdlReader.setFeature("javax.wsdl.importDocuments", false);
+
+            return wsdlReader.readWSDL(null, getSecuredParsedDocumentFromContent(wsdl));
+        } catch (Exception e) {

```

Comment:
Here we don't need to log and throw.

Reply:
fixed in e18f43f5dee9ed43a3239af62f38d67c72c78fa2